protocol/x/vault/keeper/orders_test.go

```diff
@@ -249,79 +249,167 @@ func TestGetVaultClobOrders(t *testing.T) {
+			// To calculate size of each order
+			// 1. `order_size_ppm * equity / oracle_price`.
 			expectedOrderSubticks: []uint64{
 				// spreadPpm = max(3_000, 1_500 + 50) = 3_000
-				// priceSubticks = 5_000_000_000 * 10^(-5 - (-8) + (-10) - (-6)) = 5 * 10^8
-				// a_1 = 5 * 10^8 * (1 + 0.003)^1 = 501_500_000
+				// spread = 0.003
+				// leverage = 0 / 1_000 = 0
+				// oracleSubticks = 5_000_000_000 * 10^(-5 - (-8) + (-10) - (-6)) = 5 * 10^8
+				// leverage_0 = leverage = 0
+				// skew_0 = -0 * 3_000 * 0.5 = 0
+				// a_0 = 5 * 10^8 * (1 + 0) * (1 + 0.003)^1 = 501_500_000
 				501_500_000,
-				// b_1 = 5 * 10^8 * (1 - 0.003)^1 = 498_500_000
-				498_500_000,
-				// a_2 = 5 * 10^8 * (1 + 0.003)^2 = 503_004_500
-				503_004_500,
-				// b_2 = 5 * 10^8 * (1 - 0.003)^2 = 497_004_500
-				497_004_500,
+				// b_0 = 5 * 10^8 * (1 + 0) / (1 + 0.003)^1 = 498_504_486
+				// round down to nearest multiple of subticks_per_tick=5.
+				498_504_485,
+				// leverage_1 = leverage - 0.1 = -0.1
+				// skew_1 = 0.1 * 0.003 * 0.5 = 0.00015
+				// a_1 = 5 * 10^8 * (1 + 0.00015) * (1 + 0.003)^2 = 503_079_950.675
+				// round up to nearest multiple of subticks_per_tick=5.
+				503_079_955,
+				// leverage_1 = leverage + 0.1 = 0.1
+				// skew_1 = -0.1 * 0.003 * 0.5 = -0.00015
+				// b_2 = 5 * 10^8 * (1 - 0.00015) / (1 + 0.003)^2 ~= 496_938_894.184
+				// round down to nearest multiple of subticks_per_tick=5.
+				496_938_890,
 			},
-			expectedOrderQuantums: []uint64{ // TODO (TRA-144): Implement order size
-				5,
-				5,
-				5,
-				5,
+			// order_size = 10% * 1_000 / 50_000 = 0.002
+			// order_size_base_quantums = 0.002 * 10^10 = 20_000_000
+			expectedOrderQuantums: []uint64{
+				20_000_000,
+				20_000_000,
+				20_000_000,
+				20_000_000,
 			},
 		},
 		"Success - Get orders from Vault for Clob Pair 1": {
-			vaultId:  constants.Vault_Clob_1,
-			clobPair: constants.ClobPair_Eth,
-			marketParam: pricestypes.MarketParam{
-				Id:                 constants.TestMarketParams[1].Id,
-				Pair:               constants.TestMarketParams[1].Pair,
-				Exponent:           constants.TestMarketParams[1].Exponent,
-				MinExchanges:       constants.TestMarketParams[1].MinExchanges,
-				MinPriceChangePpm:  4_200, // Set a high min price change to test spread calculation.
-				ExchangeConfigJson: constants.TestMarketParams[1].ExchangeConfigJson,
+			vaultParams: vaulttypes.Params{
+				Layers:                 3,       // 3 layers
+				SpreadMinPpm:           3_000,   // 30 bps
+				SpreadBufferPpm:        4_500,   // 15 bps
+				SkewFactorPpm:          700_000, // 0.7
+				OrderSizePpm:           50_000,  // 5%
+				OrderExpirationSeconds: 4,       // 4 seconds
 			},
-			marketPrice: constants.TestMarketPrices[1],
-			perpetual:   constants.EthUsd_20PercentInitial_10PercentMaintenance,
+			vaultId:            constants.Vault_Clob_1,
+			vaultAssetQuantums: big.NewInt(2_000_000_000), // 2,000 USDC
+			vaultInventory:     big.NewInt(-1_000),
```

Comment:
question: is inventory also in quantums?

Reply:
inventory is in base quantums

Reply:
Update naming? Or add comment noting how much of the base currency it is?

Reply:
just making sure vaultAssetQuantums could be negative and the account could still have positive equity if the vaultInventory is valued higher.

note: the accounting is done such that we make USDC on every short and spend USDC for every long.

Reply:
Yeah asset position can be negative and subaccount can still have positive equity